tests/test_keys.py

```diff
@@ -1,47 +1,46 @@
 import pytest
 
 from umbral import umbral, keys
+from umbral.params import UmbralParameters
 
 
 def test_gen_key():
-    umbral_priv_key = keys.UmbralPrivateKey.gen_key(umbral.UmbralParameters())
+    # Pass in the parameters to test that manual param selection works
+    umbral_priv_key = keys.UmbralPrivateKey.gen_key(UmbralParameters())
     assert type(umbral_priv_key) == keys.UmbralPrivateKey
 
-    umbral_pub_key = umbral_priv_key.get_pub_key(umbral.UmbralParameters())
+    umbral_pub_key = umbral_priv_key.get_pub_key(UmbralParameters())
     assert type(umbral_pub_key) == keys.UmbralPublicKey
 
 
 def test_private_key_serialization():
-    pre = umbral.PRE(umbral.UmbralParameters())
+    pre = umbral.PRE()
 
     priv_key = pre.gen_priv()
     umbral_key = keys.UmbralPrivateKey(priv_key)
 
     encoded_key = umbral_key.save_key()
 
-    decoded_key = keys.UmbralPrivateKey.load_key(encoded_key,
-                                                 umbral.UmbralParameters())
+    decoded_key = keys.UmbralPrivateKey.load_key(encoded_key)
 
     assert priv_key == decoded_key.bn_key
 
 
 def test_private_key_serialization_with_encryption():
-    pre = umbral.PRE(umbral.UmbralParameters())
+    pre = umbral.PRE()
 
     priv_key = pre.gen_priv()
     umbral_key = keys.UmbralPrivateKey(priv_key)
 
     encoded_key = umbral_key.save_key(password=b'test')
 
-    decoded_key = keys.UmbralPrivateKey.load_key(encoded_key,
-                                                 umbral.UmbralParameters(),
-                                                 password=b'test')
+    decoded_key = keys.UmbralPrivateKey.load_key(encoded_key, password=b'test')
 
     assert priv_key == decoded_key.bn_key
 
 
 def test_public_key_serialization():
-    pre = umbral.PRE(umbral.UmbralParameters())
+    pre = umbral.PRE()
 
     priv_key = pre.gen_priv()
     pub_key = pre.priv2pub(priv_key)
```

Comment:
There are two ways to generate public keys from private keys: using pre.priv2pub() and priv_key.gen_pub_key(). We should probably provide a single implementation, even if we maintain both functions.

Reply:
Of course, this is not a thing to do right now. We can add it to the TODO. 
BTW, I prefer to keep priv_key.gen_pub_key() and to remove pre.priv2pub().

Reply:
Specifically, those methods deal with `Point` and `BigNum` objects.

Yeah, it's a little weird. We could add in checks and a bool flag for the Umbral key objects there and generate accordingly.

Could also just remove them, too.